src/main/kotlin/com/kebab/user/security/service/UserDetailsServiceImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.kebab.user.security.service
+
+import com.kebab.user.repository.UserRepository
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.security.core.userdetails.UserDetailsService
+import org.springframework.security.core.userdetails.UsernameNotFoundException
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+
+@Service
+class UserDetailsServiceImpl(val userRepository: UserRepository) : UserDetailsService {
+    @Transactional(readOnly = true)
+    @Throws(UsernameNotFoundException::class)
+    override fun loadUserByUsername(username: String): UserDetails {
+        val user= userRepository.findByUsername(username) ?: com.kebab.user.model.User(username, "kekPassword")
```

Comment:
Import?

Reply:
I see, that name `User` is already reserved, we can rename our entity to `Account` or something like this

Reply:
General recommendation

Kotlin allows to craete functions without specifying exact return type

If you can not specify it, please do not do it

It will help if refactoring in future